### ISSUE TITLE
Duplicate CA1822 warnings after installing installing fxcop and building

### ISSUE THREAD

**Issue by mattscheffer** [dotnet/roslyn#36189]
**Version Used**: 
29004.99
**Steps to Reproduce**:

1. Create a new C# .net core Console app
2. Add a new class file
3. Adjust the code for the class to include:
```
public class Class1
    {
        public int Method1()
        {
            return 22;
        }
    }
```
4. Install latest Microsoft.CodeAnalysis.FxCopAnalyzers NuGet package
5. Observe CA1822 in the error list for Method1()
6. Build
**Expected Behavior**:
Only one entry for CA1822 appears
**Actual Behavior**:
2 entries appear. 1 from Intellisense and 1 from Build

![image](https://user-images.githubusercontent.com/10537189/58993012-26167080-87a1-11e9-952b-1e1a282ed44e.png)

**Comment by heejaechang** [comment]
@mavasani this is happening since the analyzer "https://github.com/dotnet/roslyn-analyzers/blob/63bf71fd17abbb4143fdc4c78d369edcdeaf9db4/src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs#L13"

implements both operation end analyzer + compilation end analyzer.

since this analyzer is compilation end analyzer and full solution analysis is off, we don't do de-dup. but it is reporting live error from operation end analyzer.

this is limitation of our dedup system. and there is no good way to figure this out when full solution analysis is off.
